src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -105,12 +125,71 @@ _IRQL_requires_max_(DISPATCH_LEVEL)
 _Function_class_(CXPLAT_DATAPATH_RECEIVE_CALLBACK)
 void
 UdpRecvCallback( 
-    _In_ CXPLAT_SOCKET* /* Binding */,
-    _In_ void* /* Context */,
+    _In_ CXPLAT_SOCKET* Binding,
+    _In_ void* Context,
     _In_ CXPLAT_RECV_DATA* RecvBufferChain
     )
 {
+    CXPLAT_RECV_DATA* Datagram;
+    while ((Datagram = RecvBufferChain) != NULL) {
+        RecvBufferChain = Datagram->Next;
+        Datagram->Next = NULL;
+        uint8_t DestCidLen, SourceCidLen;
+        const uint8_t* DestCid, *SourceCid;
+        
+        QUIC_RX_PACKET* Packet = (QUIC_RX_PACKET *)CXPLAT_ALLOC_NONPAGED(sizeof(QUIC_RX_PACKET), QUIC_POOL_TOOL);
```

Comment:
Is it possible to allocate the packet inside the do while loop and directly enqueue it without any further allocs/copies?

Reply:
Or alternatively, we could just use stack memory for this temp Packet instead of alloc/free?